Source/Core/DolphinWX/Debugger/CodeWindow.cpp

```diff
@@ -339,70 +339,20 @@ void CCodeWindow::StepOver()
   }
 }
 
-// Returns true on a rfi, blr or on a bclr that evaluates to true.
-static bool WillInstructionReturn(UGeckoInstruction inst)
-{
-  // Is a rfi instruction
-  if (inst.hex == 0x4C000064u)
-    return true;
-  bool counter = (inst.BO_2 >> 2 & 1) != 0 || (CTR != 0) != ((inst.BO_2 >> 1 & 1) != 0);
-  bool condition = inst.BO_2 >> 4 != 0 || GetCRBit(inst.BI_2) == (inst.BO_2 >> 3 & 1);
-  bool isBclr = inst.OPCD_7 == 0b010011 && (inst.hex >> 1 & 0b10000) != 0;
-  return isBclr && counter && condition && !inst.LK_3;
-}
-
 void CCodeWindow::StepOut()
 {
   if (CPU::IsStepping())
   {
     CPU::PauseAndLock(true, false);
-    PowerPC::breakpoints.ClearAllTemporary();
-
-    // Keep stepping until the next return instruction or timeout after five seconds
-    using clock = std::chrono::steady_clock;
-    clock::time_point timeout = clock::now() + std::chrono::seconds(5);
-    PowerPC::CoreMode old_mode = PowerPC::GetMode();
-    PowerPC::SetMode(PowerPC::CoreMode::Interpreter);
-
-    // Loop until either the current instruction is a return instruction with no Link flag
-    // or a breakpoint is detected so it can step at the breakpoint. If the PC is currently
-    // on a breakpoint, skip it.
-    UGeckoInstruction inst = PowerPC::HostRead_Instruction(PC);
-    do
-    {
-      if (WillInstructionReturn(inst))
-      {
-        PowerPC::SingleStep();
-        break;
-      }
-
-      if (inst.LK)
-      {
-        // Step over branches
-        u32 next_pc = PC + 4;
-        do
-        {
-          PowerPC::SingleStep();
-        } while (PC != next_pc && clock::now() < timeout &&
-                 !PowerPC::breakpoints.IsAddressBreakPoint(PC));
-      }
-      else
-      {
-        PowerPC::SingleStep();
-      }
-
-      inst = PowerPC::HostRead_Instruction(PC);
-    } while (clock::now() < timeout && !PowerPC::breakpoints.IsAddressBreakPoint(PC));
-
-    PowerPC::SetMode(old_mode);
+    bool step_out_success = PowerPC::StepOut(std::chrono::seconds(5));
```

Comment:
`const bool`

Reply:
Unless I'm mistaken I can't because it generates on MSVC:
```
INTERNAL COMPILER ERROR in 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x64\CL.exe'
```

Reply:
lmao

Reply:
The following lines added anywhere in the Dolphin code base are able to trigger the same issue:
```C++
#include <chrono>
#include <optional>

namespace {
  int Foo(std::optional<int> i) { return 21; }
  void Bar() { const int i = Foo(23); }
}
```
If the const specifier is removed or the optional class isn't used, it doesn't happen.

Reply:
Have you reported this to Microsoft yet? Sounds like something rather serious that should be addressed by them.

Reply:
I haven't yet because I was busy this morning, but I'll tonight.

Reply:
Thanks Sebastien for reporting. As mentioned to you in email, we build Dolphin sources from here “git clone https://github.com/dolphin-emu/dolphin.git D:\Dolphin\src” with 3 different compiler modes – default, /std:c++17, and /permissive-, on regular basis (daily). It's surprising that you're seeing this crash. We'll investigate into this and will keep you posted.

Reply:
This is an unmerged PR, it will (obviously) not crash in your tests since you build the `master` branch. And we will not merge it with Windows builds failing due to that; so most likely we'll just _not_ use `const` in that place until it is fixed.
But thanks for keeping us updated! :+1:

Reply:
Ok. We'll update with when we can fix the bug, and if there's any quick source workaround you may be able to apply in the meantime.